catkin_ws/src/arena_camera/src/arena_camera_node.cpp

```diff
@@ -1269,13 +1269,16 @@ void ArenaCameraNode::spin()
     }
 
     if ((getNumSubscribersRect() > 0 || iox_state_->iox_rect.hasSubscribers()) &&
-      camera_info_manager_->isCalibrated())
+      camera_info_manager_->isCalibrated() && pinhole_model_->initialized())
     {
-      cv_bridge_img_rect_->header.stamp = img_raw_msg_.header.stamp;
-      assert(pinhole_model_->initialized());
-      cv_bridge::CvImagePtr cv_img_raw = cv_bridge::toCvCopy(img_raw_msg_, img_raw_msg_.encoding);
       pinhole_model_->fromCameraInfo(camera_info_manager_->getCameraInfo());
-      pinhole_model_->rectifyImage(cv_img_raw->image, cv_bridge_img_rect_->image);
+
+      const cv::Mat img_raw = cv_bridge::toCvCopy(img_raw_msg_)->image;
+      cv::Mat img_rect{ };
```

Comment:
not sure the rest of this code assumes c++11 usage, so we may need to avoid init lists